### PR TITLE
Enh: Show mark in stream entries and comments

### PR DESCRIPTION
**Pull request by felixhahnweilheim**
I have shortly tested with posts, calendar entries, tasks and polls.

A bit more testing might be good before merging it.

### CODE REVIEW

**Comment by felixhahnweilheim** [comment]
I am not overwriting the class WallStreamEntryWidget as it is an abstract class and it would be needed to copy a lot of code.

The only disadvantage is that some logic has to be done in the view file. (I tried to make clear which parts are specific for the verified module.)

**Comment by ArchBlood** [comment]
Whenever you're finished I'll look over it for review.

File: config.php [hunk 1]
```diff
@@ -5,6 +5,7 @@
 use humhub\modules\content\widgets\ContainerProfileHeader;
 use humhub\modules\user\widgets\PeopleCard;
 use humhub\modules\space\widgets\SpaceDirectoryCard;
+use humhub\modules\content\widgets\stream\WallStreamEntryWidget
```
Comment (ArchBlood) [inline config.php:8]: Missing `;` at the end

**Comment by ArchBlood** [comment]
> I am not overwriting the class WallStreamEntryWidget as it is an abstract class and it would be needed to copy a lot of code.

You should still be able to extend `WallStreamEntryWidget` if needed and instead of using all of the code within the file just add your own logic to the new widget class;

See the following for example;
https://github.com/humhub/humhub/blob/master/protected/humhub/modules/content/widgets/stream/WallStreamModuleEntryWidget.php

On this note, we should think of using different names as adding Overwrite to each new widget makes the file names rather long, so something like the following should do;

ContainerProfileHeaderOverwrite > VerifiedProfileHeader

PeopleCardOverwrite > VerifiedPeopleCard

SpaceDirectoryCardOverwrite > VerifiedSpaceCard

Your thoughts?

**Comment by ArchBlood** [comment]
I've noticed that comment and replies do not show the tick, it's not really an issue but I'm sure that someone would be bugged about it in the long run.

**Comment by felixhahnweilheim** [comment]
Thanks for the example link but if I try it in our context I only get `yii\di\NotInstantiableException: Can not instantiate humhub\modules\verified\widgets\VerifiedStreamEntry`

Yes, I see showing the mark in comments as a seperate issue.

**Comment by felixhahnweilheim** [comment]
So now with https://github.com/GreenMeteor/verified/pull/9/commits/573aaae87b768e994b4ade138c1411c680552d98 I am happy with the PR and I don't think we need to overwrite the widget class.

I have tested with the stream entries of: posts, calendar, tasks, polls, link lists, files, gallery and wiki pages.

And I have updated the class names (should we mention it in the changelog?)

**Comment by ArchBlood** [comment]
> So now with https://github.com/GreenMeteor/verified/pull/9/commits/573aaae87b768e994b4ade138c1411c680552d98 I am happy with the PR and I don't think we need to overwrite the widget class.
> 
> I have tested with the stream entries of: posts, calendar, tasks, polls, link lists, files, gallery and wiki pages.
> 
> And I have updated the class names (should we mention it in the changelog?)
> 

Yes, we should mark this in the changelogs, we should probably also look into creating a migration for those that are using the old class names if necessary, so that no errors happen with the class name changes done.

**Comment by felixhahnweilheim** [comment]
Changelog updated.  

I don't think migrations are necessary. For the database nothing changed.

**Comment by ArchBlood** [comment]
I'll look over the p/r then merge it for the next release along with the last p/r done, this should make the users slightly happier.

#10 will be implemented in the out of beta stage then I'll work on getting the notifications implemented in the second full release.

**Comment by ArchBlood** [comment]
Looking at the comments/reply the following view is needed;
https://github.com/humhub/humhub/blob/master/protected/humhub/modules/comment/widgets/views/comment.php

**Comment by felixhahnweilheim** [comment]
Now comments/replies is also implented.

**Comment by felixhahnweilheim** [comment]
Until now the code to check if a user/space is verified and to display the icon was spread over all the widgets. So  with the last commit I introduced a new widget `VerifiedIcon` and moved the logic into it.

Now the other widget files only contain the method `getViewPath()` and in the view files the `VerifiedIcon` widget is added.

I think this will reduce future maintainance effort.

On top of that custom themes can use that widget in their views (added a hint to `README.md`, hope that is the right place).

**Comment by ArchBlood** [comment]
> Until now the code to check if a user/space is verified and to display the icon was spread over all the widgets. So  with the last commit I introduced a new widget `VerifiedIcon` and moved the logic into it.
> 
> Now the other widget files only contain the method `getViewPath()` and in the view files the `VerifiedIcon` widget is added.
> 
> I think this will reduce future maintainance effort.
> 
> On top of that custom themes can use that widget in their views (added a hint to `README.md`, hope that is the right place).

Great work! Personally I was thinking of creating the same exact thing but named `VerifyLoader` but we can go with yours.

**Comment by felixhahnweilheim** [comment]
We can still rename it if you prefer.

**Comment by ArchBlood** [comment]
> We can still rename it if you prefer.

The name is good as is.

**Comment by ArchBlood** [comment]
Are we good to merge with main branch or do we still need more testing?

**Comment by felixhahnweilheim** [comment]
You can merge it.

File: docs/README.md [hunk 2]
```diff
@@ -9,3 +9,9 @@ Verify your HumHub users with a verified tick on their profiles.
 [Green Meteor](https://github.com/GreenMeteor)
 
 [Felix Hahn](https://github.com/felixhahnweilheim)
+
+### For theme developpers
```
Comment (ArchBlood) [inline docs/README.md:13]: Maybe we can create a `/docs/DEVELOPERS.md` or `/docs/MANUAL.md` file for this?

Comment (felixhahnweilheim) [reply docs/README.md]: Ok. I have added DEVELOPER.md as mentioned [in HumHub docs](https://docs.humhub.org/docs/develop/modules#documentation).

**Pull request merged**